src/aeso_fcode_to_fate.erl

```diff
@@ -231,7 +231,28 @@ lookup_var(#env{vars = Vars}, X) ->
 
 %% -- The compiler --
 
+term_to_fate({lit, L}) ->
+     case L of
+        {int, N}             -> aeb_fate_data:make_integer(N);
+        {string, S}          -> aeb_fate_data:make_string(S);
+        {bool, B}            -> aeb_fate_data:make_boolean(B);
+        {account_pubkey, K}  -> aeb_fate_data:make_address(K);
+        {contract_pubkey, K} -> aeb_fate_data:make_contract(K);
+        {oracle_pubkey, K}   -> aeb_fate_data:make_oracle(K);
+        {oracle_query_id, _} -> ?TODO(fate_oracle_query_id_value)
+     end;
+term_to_fate(nil) ->
+    aeb_fate_data:make_list([]);
+term_to_fate({con, _Ar, I, As}) ->
+    %% what is Ar
+    aeb_fate_data:make_list([term_to_fate(I) | [ term_to_fate(A) || A<-As]]);
+term_to_fate({tuple, As}) ->
+    aeb_fate_data:make_tuple(list_to_tuple([ term_to_fate(A) || A<-As])).
```

Comment:
TODO:
- variant types
- negative integers (`{op, '-', [{lit, {int, N}}]}`)
- Bits (not working for aevm though)
- hash and signature literals (coordinate with @hanssv)

Reply:
Thanks. Will check with @hanssv. Pushed the test for dutch_auction, counter and bitcoin_auth that take different arguments. Bitcoin_auth is not compilable yet, but when it is, test is ready.

Reply:
The test would look nicer if you defined a single contract with functions whose arguments cover all the cases in `term_to_fate`. Choosing a different existing test contract and defining a new unit test for each case is a bit clumsy, and it's hard to get an overview of the test coverage.